Repository/ExtendedRepositories/PermissionsRepository.cs

```diff
@@ -42,6 +43,13 @@ public void AssignPermissionToRole(string Permission, int RoleId)
                 PermissionId = GetPermission(Permission).Id
             });
         }
+
+        public void RemovePermissionFromRole(string Permission, int RoleId)
+        {
+            var userRole = RolePermissionRepository.GetAll().Where(x => x.Permission.Name == Permission && x.Role.Id == RoleId).FirstOrDefault();
+            RolePermissionRepository.SoftDelete(userRole);

```

Comment:
Create a version that takes both arguments as string for optimization

Reply:
there will be no optimization since the name and the id are both attributes and no additional computation is done to fetch the object with the integer id

Reply:
For the Logic layer to effectively use this function, it has to fetch the role first, save this additional query by making a string only version